app/App.tsx

```diff
@@ -48,6 +120,10 @@ export const App = () => {
     });
   }, [history]);
 
+  if (!userLocation) {
+    return <Loader />;
+  }
```

Comment:
This may not be the perfect boundary to put a loader. This is something I think we need to work out a bit more with product/design in terms of setting expectations on load.